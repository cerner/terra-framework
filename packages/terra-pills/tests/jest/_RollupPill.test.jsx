import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mountWithIntl } from 'terra-enzyme-intl';
import RollupPill from '../../src/private/_RollupPill';

describe('Rollup Pill', () => {
  it('should render rollup pill with label show more', () => {
    const wrapper = mountWithIntl(<RollupPill isCollapsed rollupCount={4} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render rollup pill with label show less when isCollapsed is false', () => {
    const wrapper = mountWithIntl(<RollupPill isCollapsed={false} rollupCount={3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render rollup pill with label show less when isCollapsed is true and rollupCount is 0', () => {
    const wrapper = mountWithIntl(<RollupPill isCollapsed rollupCount={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
