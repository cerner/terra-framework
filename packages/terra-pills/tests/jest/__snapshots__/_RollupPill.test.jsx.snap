// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rollup Pill should render rollup pill with label show less when isCollapsed is false 1`] = `
<InjectIntl(RollUpPill)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isCollapsed={false}
  rollupCount={3}
>
  <RollUpPill
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isCollapsed={false}
    rollupCount={3}
  >
    <div
      className="pill-list-item"
      role="listitem"
    >
      <button
        aria-describedby="rollup-pill-hint"
        className="rollup-pill"
        data-terra-rollup-pill={true}
        data-terra-rollup-pill-show-focus-styles={true}
        id="rollup-pill"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="rollup-pill-label"
        >
          Terra.pills.label.showLess
        </span>
      </button>
      <VisuallyHiddenText
        aria-hidden="true"
        id="rollup-pill-hint"
        text="Terra.pills.hint.showLess"
      >
        <span
          aria-hidden="true"
          className="visually-hidden-text"
          id="rollup-pill-hint"
        >
          Terra.pills.hint.showLess
        </span>
      </VisuallyHiddenText>
    </div>
  </RollUpPill>
</InjectIntl(RollUpPill)>
`;

exports[`Rollup Pill should render rollup pill with label show less when isCollapsed is true and rollupCount is 0 1`] = `
<InjectIntl(RollUpPill)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isCollapsed={true}
  rollupCount={0}
>
  <RollUpPill
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isCollapsed={true}
    rollupCount={0}
  >
    <div
      className="pill-list-item"
      role="listitem"
    >
      <button
        aria-describedby="rollup-pill-hint"
        className="rollup-pill"
        data-terra-rollup-pill={true}
        data-terra-rollup-pill-show-focus-styles={true}
        id="rollup-pill"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="rollup-pill-label"
        >
          Terra.pills.label.showLess
        </span>
      </button>
      <VisuallyHiddenText
        aria-hidden="true"
        id="rollup-pill-hint"
        text="Terra.pills.hint.rollupPill"
      >
        <span
          aria-hidden="true"
          className="visually-hidden-text"
          id="rollup-pill-hint"
        >
          Terra.pills.hint.rollupPill
        </span>
      </VisuallyHiddenText>
    </div>
  </RollUpPill>
</InjectIntl(RollUpPill)>
`;

exports[`Rollup Pill should render rollup pill with label show more 1`] = `
<InjectIntl(RollUpPill)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
  isCollapsed={true}
  rollupCount={4}
>
  <RollUpPill
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
    isCollapsed={true}
    rollupCount={4}
  >
    <div
      className="pill-list-item"
      role="listitem"
    >
      <button
        aria-describedby="rollup-pill-hint"
        className="rollup-pill"
        data-terra-rollup-pill={true}
        data-terra-rollup-pill-show-focus-styles={true}
        id="rollup-pill"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="rollup-pill-label"
        >
          Terra.pills.label.rollupPill
        </span>
      </button>
      <VisuallyHiddenText
        aria-hidden="true"
        id="rollup-pill-hint"
        text="Terra.pills.hint.rollupPill"
      >
        <span
          aria-hidden="true"
          className="visually-hidden-text"
          id="rollup-pill-hint"
        >
          Terra.pills.hint.rollupPill
        </span>
      </VisuallyHiddenText>
    </div>
  </RollUpPill>
</InjectIntl(RollUpPill)>
`;
